{"url": "https://api.github.com/repos/datalad/datalad/issues/4609", "repository_url": "https://api.github.com/repos/datalad/datalad", "labels_url": "https://api.github.com/repos/datalad/datalad/issues/4609/labels{/name}", "comments_url": "https://api.github.com/repos/datalad/datalad/issues/4609/comments", "events_url": "https://api.github.com/repos/datalad/datalad/issues/4609/events", "html_url": "https://github.com/datalad/datalad/issues/4609", "id": 632334071, "node_id": "MDU6SXNzdWU2MzIzMzQwNzE=", "number": 4609, "title": "GitRepo.update_submodule() cannot handle non-master branch in submodule", "user": {"login": "mih", "id": 136479, "node_id": "MDQ6VXNlcjEzNjQ3OQ==", "avatar_url": "https://avatars.githubusercontent.com/u/136479?v=4", "gravatar_id": "", "url": "https://api.github.com/users/mih", "html_url": "https://github.com/mih", "followers_url": "https://api.github.com/users/mih/followers", "following_url": "https://api.github.com/users/mih/following{/other_user}", "gists_url": "https://api.github.com/users/mih/gists{/gist_id}", "starred_url": "https://api.github.com/users/mih/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/mih/subscriptions", "organizations_url": "https://api.github.com/users/mih/orgs", "repos_url": "https://api.github.com/users/mih/repos", "events_url": "https://api.github.com/users/mih/events{/privacy}", "received_events_url": "https://api.github.com/users/mih/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 390153891, "node_id": "MDU6TGFiZWwzOTAxNTM4OTE=", "url": "https://api.github.com/repos/datalad/datalad/labels/UX", "name": "UX", "color": "0052cc", "default": false, "description": "user experience"}, {"id": 1782620215, "node_id": "MDU6TGFiZWwxNzgyNjIwMjE1", "url": "https://api.github.com/repos/datalad/datalad/labels/severity-minor", "name": "severity-minor", "color": "ffdddd", "default": false, "description": "a problem which doesn't affect the package's usefulness, and is presumably trivial to fix"}], "state": "open", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2020-06-06T09:42:41Z", "updated_at": "2020-06-06T09:45:49Z", "closed_at": null, "author_association": "MEMBER", "active_lock_reason": null, "body": "Originally discovered by @adswa and reproducible with this:\r\n\r\n```\r\n% datalad clone https://gin.g-node.org/adswa/OSRdemo\r\n% cd OSRdemo\r\n% datalad get -n .source\r\n[INFO   ] Scanning for unlocked files (this may take some time)                              \r\n[INFO   ] Configured RIA store not found at ria+file:///ds/hcp/srv \r\n[WARNING] /tmp/OSRdemo/.source has a detached HEAD since cloned branch master has another common ancestor with 957c6a58 \r\ninstall(ok): /tmp/OSRdemo/.source (dataset) [Installed subdataset in order to get /tmp/OSRdemo/.source]\r\n```\r\n\r\n`.source` stays with a detached head.\r\n\r\nFor good reasons, the superdataset is on the `master` branch, while the submodule references a dataset state in a `bids` branch. The logic inside is therefore broken https://github.com/datalad/datalad/blob/master/datalad/support/gitrepo.py#L2941\r\n\r\nWe would need to record the target branch of a submodule in a parent to fix this issue. This is a long identified need #1424, an implementation was attempted #3817, and removed again #4375 due to problems with the approach #4373 \r\n\r\nSeverity minor, because the warning is annoying, but there is no real issue in the context of the use case above.", "closed_by": null, "reactions": {"url": "https://api.github.com/repos/datalad/datalad/issues/4609/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/datalad/datalad/issues/4609/timeline", "performed_via_github_app": null}